File: public/public-web/src/main/java/ch/difty/scipamato/publ/web/model/KeywordModel.java
package ch.difty.scipamato.publ.web.model;

import java.util.List;

import org.apache.wicket.spring.injection.annot.SpringBean;
import org.jetbrains.annotations.NotNull;

import ch.difty.scipamato.common.web.model.InjectedLoadableDetachableModel;
import ch.difty.scipamato.publ.entity.Keyword;
import ch.difty.scipamato.publ.persistence.api.KeywordService;

public class KeywordModel extends InjectedLoadableDetachableModel<Keyword> {

    private static final long serialVersionUID = 1L;

    @SpringBean
    private KeywordService service;

    private final String languageCode;

    public KeywordModel(@NotNull final String languageCode) {
        super();
        this.languageCode = languageCode;
    }

    @NotNull
    @Override
    protected List<Keyword> load() {
        return service.findKeywords(languageCode);
    }
}

File: public/public-web/src/main/kotlin/ch/difty/scipamato/publ/web/model/KeywordModel.kt
package ch.difty.scipamato.publ.web.model

import ch.difty.scipamato.common.web.model.InjectedLoadableDetachableModel
import ch.difty.scipamato.publ.entity.Keyword
import ch.difty.scipamato.publ.persistence.api.KeywordService
import org.apache.wicket.spring.injection.annot.SpringBean

class KeywordModel(
    private val languageCode: String,
) : InjectedLoadableDetachableModel<Keyword>() {

    @SpringBean
    private lateinit var service: KeywordService

    public override fun load(): List<Keyword> = service.findKeywords(languageCode)

    companion object {
        private const val serialVersionUID = 1L
    }
}
